driver/ni/task.cpp

```diff
@@ -183,179 +193,163 @@ void ni::ReaderTask::exec(task::Command &cmd) {
+                           synnax::Task task,
+                           std::shared_ptr<pipeline::Sink> sink,
+                           std::shared_ptr<ni::DigitalWriteSink> ni_sink,
+                           std::shared_ptr<pipeline::Source> state_source,
+                           synnax::WriterConfig writer_config,
+                           synnax::StreamerConfig streamer_config,
+                           const breaker::Config breaker_config
 ) : ctx(ctx),
     task(task),
-    cmd_write_pipe(pipeline::Control(ctx->client, streamer_config, std::move(sink), breaker_config)),
-    state_write_pipe(pipeline::Acquisition(ctx->client, writer_config, state_source, breaker_config)){
+    cmd_write_pipe(pipeline::Control(ctx->client, streamer_config, std::move(sink),
+                                     breaker_config)),
+    state_write_pipe(
+        pipeline::Acquisition(ctx->client, writer_config, state_source,
+                              breaker_config)),
+    sink(ni_sink) {
 }
 
 
-std::unique_ptr <task::Task> ni::WriterTask::configure(const std::shared_ptr <task::Context> &ctx,
-                                                       const synnax::Task &task) {
+std::unique_ptr<task::Task> ni::WriterTask::configure(
+    const std::shared_ptr<task::Context> &ctx,
+    const synnax::Task &task) {
     // create a breaker config TODO: use the task to generate the other parameters?
     auto breaker_config = breaker::Config{
-            .name = task.name,
-            .base_interval = 1 * SECOND,
-            .max_retries = 20,
-            .scale = 1.2,
+        .name = task.name,
+        .base_interval = 1 * SECOND,
+        .max_retries = 20,
+        .scale = 1.2,
     };
 
-    LOG(INFO) << "[NI Task] configuring task " << task.name;
+    auto parser = config::Parser(task.config);
+    auto data_saving = parser.optional<bool>("data_saving", true);
 
     TaskHandle task_handle;
-    // create a daq reader to provide to cmd read pipe as sink
     ni::NiDAQmxInterface::CreateTask("", &task_handle);
-    //print taskhandle
-    LOG(INFO) << "Task handle: " << task_handle;
 
-    auto daq_writer = std::make_unique<ni::DigitalWriteSink>(task_handle, ctx, task);
-    if (!daq_writer->ok()) {
-        LOG(ERROR) << "[NI Writer] failed to construct reader for" << task.name;
-        return nullptr;
-    }
+    LOG(INFO) << "[ni.writer] configuring task " << task.name;
+    auto daq_writer = std::make_shared<ni::DigitalWriteSink>(task_handle, ctx, task);
 
     // construct writer config
-    std::vector <synnax::ChannelKey> cmd_keys = daq_writer->getCmdChannelKeys();
-    std::vector <synnax::ChannelKey> state_keys = daq_writer->getStateChannelKeys();
+    std::vector<synnax::ChannelKey> cmd_keys = daq_writer->get_cmd_channel_keys();
+    std::vector<synnax::ChannelKey> state_keys = daq_writer->get_state_channel_keys();
 
     // create a writer config to write state channels
```

Comment:
Somewhat useless comment. This would be better represented in the variable name i.e. `state_writer_cfg`